src/main/scala/scala/xml/quote/internal/Liftables.scala

```diff
@@ -11,7 +11,30 @@ trait Liftables {
 
   val args: List[Tree]
 
-  val sx = q"_root_.scala.xml"
+  private val sx = q"_root_.scala.xml"
+  private val scopeId = q"$$scope$$"
+
+  /** When we lift, we don't know if we are within an enclosing xml element
+    * which defines a scope. In some cases we will have to fix the scope.
+    *
+    * E.g:
+    * {{{
+    *   xml"""<a xmlns:pre="scope0">${ xml"<b/>" }</a>"""
+    * }}}
+    * Here the scope of `b` is `TopScope` but should be `scope0`
+    */
+  def fixScopes(tree: Tree): Tree = {
+    var isTopScope = true
+    c.internal.typingTransform(tree)((tree, api) => tree match {
+      case q"$sx.TopScope" if !isTopScope =>
+        api.typecheck(scopeId)
```

Comment:
Before fixing the tree:
```scala
{
  val $tmpscope = _root_.scala.xml.NamespaceBinding("pre", "scope0", _root_.scala.xml.TopScope);
  {
    val $scope$ = $tmpscope;
    _root_.scala.xml.Elem(((null): String), "a", _root_.scala.xml.Null, $scope$, false, (({
      val $buf = new _root_.scala.xml.NodeBuffer();
      $buf.&+(((scala.xml.Elem.apply(((null): scala.Predef.String), "b", scala.xml.Null, scala.xml.TopScope, true, (({
        val $buf = new scala.xml.NodeBuffer();
        $buf
      }): _*))): scala.xml.Node));
      $buf
    }): _*))
  }
}
```
after fixing the tree
```scala
[error] /Users/renucci/Documents/Projects/scala-xml-quote/src/test/scala/ConstructionSuite.scala:106: not found: value $scope$
[error]     val xml1 =
[error]         ^
{
  val $tmpscope = _root_.scala.xml.NamespaceBinding("pre", "scope0", $scope$);
  {
    val $scope$ = $tmpscope;
    _root_.scala.xml.Elem(((null): String), "a", _root_.scala.xml.Null, $scope$, false, (({
      val $buf = new _root_.scala.xml.NodeBuffer();
      $buf.&+(((scala.xml.Elem.apply(((null): scala.Predef.String), "b", scala.xml.Null, $scope$, true, (({
        val $buf = new scala.xml.NodeBuffer();
        $buf
      }): _*))): scala.xml.Node));
      $buf
    }): _*))
  }
}
```
The transformed tree looks correct to me